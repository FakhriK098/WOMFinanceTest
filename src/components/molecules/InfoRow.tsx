import React from 'react'
import { StyleSheet, View } from 'react-native'
import Text from '../atoms/Text'
import { typography } from '@theme/typography'
import { colors } from '@theme/colors'

type Props = {
  label: string
  value: string | number
  isDark: boolean
}

const InfoRow = ({ label, value, isDark }: Props) => {
  return (
    <View style={styles.row}>
      <Text size={typography.sm} color={isDark ? colors.text.mutedDark : colors.text.mutedLight}>{label}</Text>
      <Text size={typography.sm} color={isDark ? colors.text.dark : colors.text.light}>{String(value)}</Text>
    </View>
  )
}

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingVertical: 6,
  },
})

export default React.memo(InfoRow)
