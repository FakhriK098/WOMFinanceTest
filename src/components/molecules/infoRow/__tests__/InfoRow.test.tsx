import React from 'react';
import renderer, { act } from 'react-test-renderer';

import InfoRow from '../InfoRow';

describe('InfoRow molecule', () => {
  it('matches snapshot and shows label/value', () => {
    let tr: renderer.ReactTestRenderer;
    act(() => {
      tr = renderer.create(<InfoRow label="Height" value={7} isDark={false} />);
    });
    expect(tr!.toJSON()).toMatchSnapshot();
    const label = tr!.root.find((inst) => inst.props && inst.props.children === 'Height');
    const value = tr!.root.find((inst) => inst.props && inst.props.children === '7');
    expect(label).toBeTruthy();
    expect(value).toBeTruthy();
    act(() => tr!.unmount());
  });
});

