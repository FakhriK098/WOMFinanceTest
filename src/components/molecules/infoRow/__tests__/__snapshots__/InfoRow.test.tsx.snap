// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoRow molecule matches snapshot and shows label/value 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingVertical": 6,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#6B7280",
          "fontSize": 14,
          "fontWeight": "400",
        },
        undefined,
      ]
    }
  >
    Height
  </Text>
  <Text
    style={
      [
        {
          "color": "#111827",
          "fontSize": 14,
          "fontWeight": "400",
        },
        undefined,
      ]
    }
  >
    7
  </Text>
</View>
`;
